### PR TITLE
feat: segmentation webview

### PR DESCRIPTION
**Pull request by jeffersonrabb**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This branch introduces `SegmentationPreview`, a new component in the Campaigns Wizard. The component combines logic from the `Preview` view and the `useStateWithPersistence` utility function, which will make it easier and more semantically clear to add the "view-as-segment" feature in multiple places. This also has the slight advantage of setting `localStorage` variables just before previewing, and unsetting them after the preview has been closed. 

### How to test the changes in this Pull Request:

This is a refactor, there should be no visible changes.

1. Verify the Preview feature in Campaigns Wizard works exactly as it did before. Test with no segment, segment set, segment + single campaign group, segment + multiple campaign groups.
2. Verify the `WebPreview` works normally in other contexts, e.g. the Site Design wizard.
3. Bonus: check localStorage before/during/after previewing as a segment. `newspack_campaigns-preview-segmentId` and `newspack_campaigns-preview-groupTaxIds` should be present only while the window is opening. Note that if the page is exited without closing the preview window the keys won't be purged, but this isn't much of a concern.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: assets/wizards/popups/components/segmentation-preview/index.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+ */
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * Internal dependencies.
+ */
+import { WebPreview } from '../../../../components/src';
+
+const SegmentationPreview = props => {
+	const postPreviewLink = window?.newspack_popups_wizard_data?.preview_post;
```
Comment (dkoo) [inline assets/wizards/popups/components/segmentation-preview/index.js:17]: This is the first time I've seen [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) in action. Always something new to learn!

File: assets/wizards/popups/components/segmentation-preview/index.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+		const params = {
+			view_as: [
+				`groups:${ campaignGroups.join( ',' ) }`,
+				...( segment.length ? [ `segment:${ segment }` ] : [] ),
+			].join( ';' ),
+		};
+		return addQueryArgs( urlToDecorate, params );
+	};
+
+	const onWebPreviewLoad = iframeEl => {
```
Comment (dkoo) [inline assets/wizards/popups/components/segmentation-preview/index.js:37]: This correctly passes the params from page to page when I navigate within the preview window. I had trouble with this working before, so this is great.

**Review by dkoo — APPROVED** [review]
Tested, works as described, and actually improves the preview behavior in my local environment.

File: assets/wizards/popups/components/segmentation-preview/index.js (hunk 1 above)
Comment (jeffersonrabb) [reply assets/wizards/popups/components/segmentation-preview/index.js:17]: I cannot claim credit :-), that's a lift from @adekbadek's [original Preview work](https://github.com/Automattic/newspack-plugin/pull/741/files#diff-b103111ab76a5a84285789bd2ae2cc0a5d4f29cdb5278802e195115d58cee273R30-R31)...

**Comment by adekbadek** [comment]
@jeffersonrabb 

> This also has the slight advantage of setting localStorage variables just before previewing, and unsetting them after the preview has been closed.

The persistence was intentionally set up so that the chosen segment and group would still be there after the user has navigated away or reloaded the page. 
With the changes here, the localStorage values are never actually read, so why should they be set? If the persistence is not needed, then `beforeLoad` & `onClose` should be removed.

**Pull request merged**